modules/amrfinder_plus.nf

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env nextflow
+nextflow.enable.dsl=2
+
+process AMRFINDER_PLUS {
+    publishDir "$launchDir/$params.outdir/func_annot"
+
+    cpus 4
+    memory { 8.GB * task.attempt }
+    errorStrategy 'retry' 
+    maxRetries 3
+
+    container 'quay.io/biocontainers/ncbi_amrfinderplus:3.11.4'
+
+    input:
+        path fna
+        path faa
+        path gff
+
+    output:
+        path("amrfinderplus.tsv"), emit: amrfinder_tsv
+
+    script:
+    if(fna.size() > 0)
+        """    
+        amrfinder --plus \
+        -n ${fna} \
+        -p ${faa} \
+        -g ${gff} \
+        -d ${params.amrfinder_plus_db} \
+        -a prokka \
+        --output amrfinderplus.tsv \
+        --threads ${task.cpus}
+        """
+    else
+        """
+        echo 'AMRFinderPlus output file empty due to empty input... generating dummy files'
+        touch amrfinderplus.tsv
```

Comment:
Maybe this should rely in the `[when](https://www.nextflow.io/docs/latest/process.html#when)` block instead?

Reply:
I am having problems using the `when` directive on the suggested modules. 

```
ERROR ~ Error executing process > 'AMRFINDER_REPORT (1)'

Caused by:
  No signature of method: nextflow.processor.TaskPath.file() is applicable for argument types: () values: []
Possible solutions: size(), size(), find(), find(groovy.lang.Closure), toFile(), isFile([Ljava.nio.file.LinkOption;) -- Check script '/hps/nobackup/rdf/metagenomics/service-team/projects/finding-pheno/mett_tl/software/momofy/./modules/amrfinder_report.nf' at line: 17


When block:
  amrfinder_tsv.file()

```
In [this](https://github.com/nextflow-io/nextflow/issues/2518) discussion, the use of `when` directive is not recommended anymore on DSL2. I'll go back to `if` statements